src/main/java/com/dtolabs/rundeck/plugin/overthere/OTWinRMFileCopier.java

```diff
@@ -0,0 +1,805 @@
+package com.dtolabs.rundeck.plugin.overthere;
+
+import com.dtolabs.rundeck.core.Constants;
+import com.dtolabs.rundeck.core.common.Framework;
+import com.dtolabs.rundeck.core.common.INodeEntry;
+import com.dtolabs.rundeck.core.dispatcher.DataContextUtils;
+import com.dtolabs.rundeck.core.execution.ExecutionContext;
+import com.dtolabs.rundeck.core.execution.service.DestinationFileCopier;
+import com.dtolabs.rundeck.core.execution.service.FileCopierException;
+import com.dtolabs.rundeck.core.execution.workflow.steps.FailureReason;
+import com.dtolabs.rundeck.core.execution.workflow.steps.StepFailureReason;
+import com.dtolabs.rundeck.core.plugins.Plugin;
+import com.dtolabs.rundeck.core.plugins.configuration.Describable;
+import com.dtolabs.rundeck.core.plugins.configuration.Description;
+import com.dtolabs.rundeck.core.plugins.configuration.PropertyUtil;
+import com.dtolabs.rundeck.core.plugins.configuration.StringRenderingConstants;
+import com.dtolabs.rundeck.core.storage.ResourceMeta;
+import com.dtolabs.rundeck.plugins.util.DescriptionBuilder;
+import com.dtolabs.rundeck.plugins.util.PropertyBuilder;
+import com.xebialabs.overthere.ConnectionOptions;
+import com.xebialabs.overthere.OverthereConnection;
+import com.xebialabs.overthere.OverthereFile;
+import com.xebialabs.overthere.RuntimeIOException;
+import com.xebialabs.overthere.cifs.CifsConnectionBuilder;
+import com.xebialabs.overthere.cifs.CifsConnectionType;
+import com.xebialabs.overthere.cifs.WinrmHttpsCertificateTrustStrategy;
+import com.xebialabs.overthere.cifs.WinrmHttpsHostnameVerificationStrategy;
+import com.xebialabs.overthere.cifs.winrm.WinRmRuntimeIOException;
+import com.xebialabs.overthere.util.DefaultAddressPortMapper;
+import org.apache.commons.compress.utils.IOUtils;
+import org.rundeck.storage.api.Path;
+import org.rundeck.storage.api.PathUtil;
+import org.rundeck.storage.api.StorageException;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.Map;
+
+import static com.xebialabs.overthere.ConnectionOptions.*;
+import static com.xebialabs.overthere.OperatingSystemFamily.WINDOWS;
+
+/**
+ * Executes a command via WinRM. Subclass to extend, there are places to alter the generation of user@realm, and to
+ * alter the password resolution mechanism.
+ * See {@link #getKerberosUsername(com.dtolabs.rundeck.plugin.overthere.OTWinRMFileCopier.ConnectionOptionsBuilder)}
+ * and
+ * {@link #getClearAuthPassword(com.dtolabs.rundeck.plugin.overthere.OTWinRMFileCopier.ConnectionOptionsBuilder)}.
+ *
+ * The connection options can be altered as well by overriding {@link #willUseConnectionOptions(ConnectionOptions)}.
+ */
+@Plugin(name = OTWinRMFileCopier.SERVICE_PROVIDER_TYPE, service = "FileCopier")
+public class OTWinRMFileCopier implements DestinationFileCopier, Describable {
+    public static final String SERVICE_PROVIDER_TYPE = "overthere-winrm";
```

Comment:
The first ~150 lines of this file are exactly the same as in OTWinRMNodeExecutor.java. Move them to a dedicated class,  containing only static infos, and use it in both classes